src/App.tsx

```diff
@@ -15,48 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'default',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+interface SortFields {
+  sortField: SortType,
+  isReverse: boolean,
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReverse } :SortFields,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField !== SortType.Default) {
+    preparedGoods.sort((goodA, goodB) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return goodA.localeCompare(goodB);
+
+        case SortType.Length:
+          return goodA.length - goodB.length;
+
+        default:
+          return 0;
```

Comment:
In that case sort method would be called even if it's not needed.
Consider moving `switch...case` outside of the sort callback